src/sql/workbench/services/connection/browser/connectionDialogService.ts

```diff
@@ -512,9 +512,25 @@ export class ConnectionDialogService implements IConnectionDialogService {
 				}, 10);
 				return;
 			}));
-
 		}
+
 		this._logService.error(message);
-		this._errorMessageService.showDialog(severity, headerTitle, message, messageDetails, actions);
+
+		// Set instructionText for MSSQL Provider Encryption error code -2146893019 thrown by SqlClient when certificate validation fails.
+		let instructionText: string;
+		if (errorCode === -2146893019) {
+			let enableTrustServerCert = localize('enableTrustServerCertificate', "Enable Trust server certificate");
+			instructionText = localize('trustServerCertInstructionText', `Encryption was enabled on this connection, review your SSL and certificate configuration for the target SQL Server, or enable 'Trust server certificate' in the connection dialog.
+
+			Do you want to enable 'Trust server certificate' on this connection and retry?`);
```

Comment:
This should probably be making clearer that trusting the server certificate is not a recommended practice security-wise (even with just a link to docs explaining the different settings and what they mean).

Wouldn't disabling encryption be the "better" option in this case? Trust server certificate is just giving a false sense of security and doesn't seem like something we should be encouraging. 

(and in general having a link to open up docs would be great since it's quite likely we'll be having a lot of people hitting this now with the new encrypt default and not knowing what any of this means)

Reply:
Disabling encryption keeps the communication channel unencrypted whereas letting Encrypt=true and enabling trust on server certificate is much secure as it will encrypt the data stream and any network communication will not happen in plain text.

Reply:
@erinstellato-ms do you have additional thoughts on this? If that should be of any concern?

Reply:
I guess that depends on your definition of "secure". If I have an encrypted stream but the endpoint is compromised (and I'm ignoring any errors with the certs being invalid) then that's still a Very Bad Thing and not a practice that should be recommended.

Reply:
In that case encrypt=false doesn't make it any secure either...

Reply:
Considering that, does updating instruction text to below look better?

![image](https://user-images.githubusercontent.com/13396919/199107125-79dbcaf7-0bcd-4799-ab97-9255ca3ef65d.png)

Reply:
Definitely better! I would love to see a link to official docs though as well in case people want to read up more about what all this means.